packages/framework/client-logger/app-insights-logger/src/fluidAppInsightsLogger.ts

```diff
@@ -76,6 +143,22 @@ export class FluidAppInsightsLogger implements ITelemetryBaseLogger {
 				filters: [],
 			},
 		};
+
+		// Deep copy filters to internal array
+		this._filters =
+			this.config.filtering.filters?.map((filter) => structuredClone(filter)) ?? [];
```

Comment:
Let's copy the whole structure, not just the filters. Right now if the user changes the mode on the config object they passed in, that would have a runtime effect because `shouldSendEvent()` cehcks `this.config.filtering.mode`.

Reply:
Updated to deep copy the entire config.